keyless/pepper/service/src/lib.rs

```diff
@@ -357,13 +358,25 @@ async fn process_common(
         .kid
         .ok_or_else(|| BadRequest("missing kid in JWT".to_string()))?;
 
-    let sig_pub_key = jwk::cached_decoding_key(&claims.claims.iss, &key_id)
+    let cached_key = jwk::cached_decoding_key_as_rsa(&claims.claims.iss, &key_id);
```

Comment:
I hope the normal (non-federated) JWKs in this cache expire eventually?

Reply:
we use a thread to update traditional JWKs every 10 sec, if that's what you are asking.

Reply:
I see. So "the cache" expires every 10 seconds then.

Reply:
I actually want to update this code with [RemoteJWK](https://docs.rs/jwtk/latest/jwtk/jwk/struct.RemoteJwksVerifier.html) which is used in API Gateway.  Is much cleaner.